pkg/grpc/universalapi/api_workflow.go

```diff
@@ -50,175 +47,196 @@ func (a *UniversalAPI) GetWorkflowAlpha1(ctx context.Context, in *runtimev1pb.Ge
+		return &runtimev1pb.StartWorkflowResponse{}, err
 	}
-	ret := &runtimev1pb.WorkflowReference{
+	ret := &runtimev1pb.StartWorkflowResponse{
 		InstanceId: resp.InstanceID,
 	}
 	return ret, nil
 }
 
 // TerminateWorkflowAlpha1 is the API handler for terminating a workflow
-func (a *UniversalAPI) TerminateWorkflowAlpha1(ctx context.Context, in *runtimev1pb.WorkflowActivityRequest) (*runtimev1pb.WorkflowActivityResponse, error) {
-	var method func(context.Context, *workflows.WorkflowReference) error
-	if in.WorkflowComponent != "" {
-		if wf, ok := a.CompStore.GetWorkflow(in.WorkflowComponent); ok {
-			method = wf.Terminate
-		}
+func (a *UniversalAPI) TerminateWorkflowAlpha1(ctx context.Context, in *runtimev1pb.TerminateWorkflowRequest) (*emptypb.Empty, error) {
+	emptyResponse := &emptypb.Empty{}
+	if err := a.validateInstanceID(in.InstanceId, false /* isCreate */); err != nil {
+		a.Logger.Debug(err)
+		return emptyResponse, err
 	}
-	return a.workflowActivity(ctx, in, method, messages.ErrTerminateWorkflow)
-}
 
-func (a *UniversalAPI) RaiseEventWorkflowAlpha1(ctx context.Context, in *runtimev1pb.RaiseEventWorkflowRequest) (*runtimev1pb.RaiseEventWorkflowResponse, error) {
-	if in.InstanceId == "" {
-		err := messages.ErrMissingOrEmptyInstance
+	workflowComponent, err := a.getComponent(in.WorkflowComponent)
+	if err != nil {
 		a.Logger.Debug(err)
-		return &runtimev1pb.RaiseEventWorkflowResponse{}, err
+		return emptyResponse, err
 	}
 
-	if in.EventName == "" {
-		err := messages.ErrMissingWorkflowEventName
+	req := &workflows.TerminateRequest{
+		InstanceID: in.InstanceId,
+	}
+	if err := workflowComponent.Terminate(ctx, req); err != nil {
+		err = messages.ErrTerminateWorkflow.WithFormat(in.InstanceId, err)
 		a.Logger.Debug(err)
-		return &runtimev1pb.RaiseEventWorkflowResponse{}, err
+		return emptyResponse, err
 	}
+	return emptyResponse, nil
+}
 
-	if in.WorkflowComponent == "" {
-		err := messages.ErrNoOrMissingWorkflowComponent
+func (a *UniversalAPI) RaiseEventWorkflowAlpha1(ctx context.Context, in *runtimev1pb.RaiseEventWorkflowRequest) (*emptypb.Empty, error) {
+	emptyResponse := &emptypb.Empty{}
+	if err := a.validateInstanceID(in.InstanceId, false /* isCreate */); err != nil {
 		a.Logger.Debug(err)
-		return &runtimev1pb.RaiseEventWorkflowResponse{}, err
+		return emptyResponse, err
 	}
 
-	workflowComponent, ok := a.CompStore.GetWorkflow(in.WorkflowComponent)
-	if !ok {
-		err := messages.ErrWorkflowComponentDoesNotExist.WithFormat(in.WorkflowComponent)
+	if in.EventName == "" {
+		err := messages.ErrMissingWorkflowEventName
 		a.Logger.Debug(err)
-		return &runtimev1pb.RaiseEventWorkflowResponse{}, err
+		return emptyResponse, err
+	}
+
+	workflowComponent, err := a.getComponent(in.WorkflowComponent)
+	if err != nil {
+		a.Logger.Debug(err)
+		return emptyResponse, err
 	}
 
 	req := workflows.RaiseEventRequest{
 		InstanceID: in.InstanceId,
 		EventName:  in.EventName,
-		Input:      in.Input,
+		EventData:  in.Input,
```

Comment:
should this be changed to `EventData` in the incoming request as well?

Reply:
Done.